docs/sensors/filters/ctx.md

```diff
@@ -0,0 +1,84 @@
+
+# Context Filter
+
+Context filter is applied to the event context. A CloudEvent from Webhook event-source has payload structure as:
+
+```json
+{
+    "context": {
+      "type": "type_of_event_source",
+      "specversion": "cloud_events_version",
+      "source": "name_of_the_event_source",
+      "id": "unique_event_id",
+      "time": "event_time",
+      "datacontenttype": "type_of_data",
+      "subject": "name_of_the_configuration_within_event_source"
+    },
+    "data": {
+      "header": {},
+      "body": {},
+    }
+}
+```
+
+## Fields
+
+Context filter has following fields:
+
+```yaml
+filters:
+  context:
+    type: event_type
+    subject: event_type
+    source: event_source
+    datacontenttype: event_type
+```
+
+You can also specify id, specversion and time fields in the YAML manifest, but they are ignored.
+
+## How it works
+
+Context filter takes a `start` and `stop` time in `HH:MM:SS` format in UTC.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: Sensor
+metadata:
+  name: with-ctx-filter
+spec:
+  template:
+    serviceAccountName: operate-workflow-sa
+  dependencies:
+    - name: test-dep
+      eventSourceName: webhook
+      eventName: example
+      filters:
+        context:
+          source: custom-webhook
+```
+
+## Practical example
+
+1. Create a webhook event-source
+
+  ```bash
+  kubectl -n argo-events apply -f https://raw.githubusercontent.com/argoproj/argo-events/stable/examples/event-sources/webhook.yaml
+  ```
+
+1. Create a webhook sensor with context filter
+
+  ```bash
+  kubectl -n argo-events apply -f https://raw.githubusercontent.com/argoproj/argo-events/stable/examples/sensors/filter-with-context.yaml
```

Comment:
I like how easy these examples are to run, but users might appreciate seeing the `yaml` inline too.

BTW, this is a 404.

Reply:
@fentonfentonfenton I see your point but at the same time it would make the whole document way too long.
If you want to see the yaml, it's only a matter of downloading it with the URL...

Reply:
👍  that `404` though...
![image](https://user-images.githubusercontent.com/13979377/149133708-85659974-41ca-4c06-a6e8-6c899ada7188.png)
🤖

Reply:
Oh i see this is in included in the PR `filter-with-context.yaml`, my bad.